File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/inventory/exchanges/ExchangeMountStableRemoveMessage.scala





















// Generated on 05/03/2013 10:34:54
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait ExchangeMountStableRemoveMessage extends MessageDeserializer {
  val messageId = 5964

  protected trait Message extends BaseMessage with Serializable {
    val messageId = ExchangeMountStableRemoveMessage.messageId

    val mountId: Double

    def serialize(buf: Buffer) {
      buf.writeDouble(mountId)
    }

  }

  protected def create(mountId: Double): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val mountId = buf.readDouble()

    create(mountId)
  }

}

object ExchangeMountStableRemoveMessage extends ExchangeMountStableRemoveMessage {
  case class MessageImpl(mountId: Double) extends Message
  type Target = MessageImpl

  override def create(mountId: Double) = MessageImpl(mountId)

  def apply(mountId: Double) = create(mountId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/ExchangeMountStableRemoveMessage.java


// Generated on 05/08/2013 19:37:54
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ExchangeMountStableRemoveMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5964;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public double mountId;
    
    public ExchangeMountStableRemoveMessage() { }
    
    public ExchangeMountStableRemoveMessage(double mountId) {
        this.mountId = mountId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeDouble(mountId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        mountId = buf.readDouble();
    }
    
}
